### ISSUE TITLE
First try with TROX

### ISSUE THREAD

**Issue by spjakob** [eriknn/ha-modbus_tcpip#1]
Don't have the time to try, but could not resist...... :) Success!

Need to see if I actually have a "problem" as indicated or not...
As I mentioned in the trox integration, would be super nice to be able to change flow unit, even if it is just when creating the device. 

Based on this very first release, I think and hope that this could be accepted into mainline HA, it's soooo much better solution than the current solution with a BIG yaml configuration file.... Configuration is a mess and even if you make this ok, you end up with lot of entities without devices. This solves both issues and I think that adding a new device here is almost as easy as current integration. :) :)

![image](https://github.com/user-attachments/assets/f12821ca-db70-4a78-97d3-6d3429a5b293)

**Comment by spjakob** [comment]
Some thoughts about future enhancements/developments.
* Since you also maintain a nice repository with brands and logos,  I think it would be nice to have this displayed for each device (like your Swegon integration do). You have probably planned this allready... 
* Option to select what sensors to read/expose via some checkboxes. Some devices (like my eastron sdm630, really have a lot of values that I do not care about, but I guess others could care about and I think it would make sense for an integration of this to "support all" but give the user the control to decide what they actually want to see/use...

Some other random ideas for future developments: 
* Very minor thing, but would be nice to be able to set/change default values for "ip" & "port", since these will be same when adding all devices... or maybe add "clone device" option. I have 10+ temperature sensors and off course I only need to add these once, but it would be a nice feature. 
* IDK about this one, maybe better solved with other existing integrations; add a schematic drawing for a device and "show values in right places" in this drawing. I think you have already done this with your casa unit... 
* I do not care so much about this, but maybe support for other languages for actual device "texts"....

**Comment by eriknn** [comment]
Great! But flowrate 9% at 136 m3/h, can that be correct? The status=problem has to be verified as well… With the Swegon driver, I was able to display the different alarms as attributes under the entity. I hope I can do this with this driver as well - but doing stuff like this is of course alot more difficult when it’s a general driver.
I’ve added the Swegon driver as well, so that’ll be availiable during this weekend I think.
Won’t do a release yet though, I need to make a better device/model selection first, and changes there will require re-add of devices…

**Comment by spjakob** [comment]
Flowrate is correct. This value is % of nominal value and nominal value for this VAV is 1513/m3/h (printed on the unit). 
Looking forward to new updates... For now I use all this in parallel with my old modbus integration so it's easy to remove and add devices without causing any issues with rest of system. 
I actually tried to make a fork and add my first device, but I realize it's been so long time since I coded or used GH that I have to read the docs for this... 

Anyway, if you want to add a very simple device (could be used as example template) that only has two values here is my configuration for an RS-WS-N01-8 Temperature and moisture sensor (generic implementation from china):

########
# TH-050
########
#
      - name: "THM50 - Humidity" 
        slave: 50
        scan_interval: 60
        address: 0
        data_type: int16
        input_type: input
        unit_of_measurement: "%"
        device_class: humidity
        scale: 0.1
        precision: 1

      - name: "THM50 - Temperature" 
        slave: 50
        scan_interval: 60
        address: 1
        data_type: int16
        input_type: input
        precision: 1
        unit_of_measurement: °C
        device_class: temperature
        scale: 0.1

**Comment by eriknn** [comment]
This is an RTU device?

**Comment by spjakob** [comment]
Yes: https://youtu.be/tgLTkWAz5Is?si=J7R2Pg1hN5Y--PSW

But accessed through my modbus ip gateway like everything else on my modbus
network...

**Comment by eriknn** [comment]
Okay, great. We can probably support RTU directly, but don’t know how common that’d be… Most people don’t have rs485-serial adapters like me I guess :)

**Comment by spjakob** [comment]
For me there is no need for direct rtu access, since I access everything
through my gateway (same like swegon and trox), but I think a few  ppl
still use direct rtu from their ha servers via adapters...

**Comment by eriknn** [comment]
Try the latest version. Still not release-able, but a lot of improvements and added devices!

**Comment by spjakob** [comment]
Great work at great speed! :) :) 

Tried all three devices:
![image](https://github.com/user-attachments/assets/09476782-59e6-4376-a5dd-d4cf1adf164d)

Temp sensor:
![image](https://github.com/user-attachments/assets/24cab194-caa5-4812-aac8-225073451395)

Trox:
![image](https://github.com/user-attachments/assets/d97a2ae2-aa0a-4173-9bff-836b53beb9cd)


Casa:
![image](https://github.com/user-attachments/assets/b0f7e8c9-5f8b-4bbc-a69a-28cb05609767)


Some quick findings/feedback.

* Pressing "Configure" on an added/existing device gives error.
* Scaling issue is back with casa temperature (R15).
* This is not an error, but with trox "Q Min (120)" is used to control "fixed flow" with an actual value (l/s, m3/h, etc.)  written to 120. This can also be achived with parameter 105 (that you now display, but this is then a "percent of nominal flow"), which I think is a less usefull (and probably less used) way to control flow. 

* Would it be possible to have a "force update" button for a device? That could be really nice if you for some reason would like to make sure you have current values even if "slow update" has not run yet. I could for example change some setting (speed etc.) on my casa unit and then would all my trox make changes, due to this external event, and it would then be nice to do a "force update" if I really want to see what is going on.... An even better idea would actually be not to do a "force update", but instead switching to "fast mode" for 1-2 minutes? 



* For trox; the displayed m3/h value (140)  is correct since I have about 39l/s now when I check manually:
![image](https://github.com/user-attachments/assets/60423529-8cbb-4d1d-86ce-c90162475b96)
...and 140/3.6 is ~38

Since I REALLY wanted this to be in l/s I checked code and then got a bit confused:

        self.Datapoints[self.GROUP_SENSORS] = {
            "Position": ModbusDatapoint(Address=4, Scaling=0.01, DataType=ModbusSensorData(units=PERCENTAGE)),
            "Position Degrees": ModbusDatapoint(Address=5, DataType=ModbusSensorData(units=DEGREE)),
            "Flowrate Percent": ModbusDatapoint(Address=6, Scaling=0.01, DataType=ModbusSensorData(units=PERCENTAGE)),
            "Flowrate Actual": ModbusDatapoint(Address=7, Scaling=1, DataType=ModbusSensorData(units=UnitOfVolumeFlowRate.CUBIC_METERS_PER_HOUR, icon="mdi:weather-windy")),
            "Analog Setpoint": ModbusDatapoint(Address=8, Scaling=0.001, DataType=ModbusSensorData(units=UnitOfElectricPotential.VOLT)),
        }

I can not see that you convert the l/s value that you should get from polling register 7 to m3/h. I'm sure it must be done somewhere, because value is displayed correctly.... :)

**Comment by spjakob** [comment]
I think "Problem" displayed for trox is wrong. I check this register and it looks like bit 15 is set:
![image](https://github.com/user-attachments/assets/52c62eec-f6b3-4eec-93c5-79e4601d47d9)

...and this bit seems to be "for internal purposes only", not a real error.

![image](https://github.com/user-attachments/assets/b76fa02a-212f-47d6-afe1-e10b8aef0c29)

**Comment by eriknn** [comment]
Try now:

`Pressing "Configure" on an added/existing device gives error.`
Many reasons this could happen, most likely just happened because of some change that was incompatible with current configuration values. Happens when I change so much stuff :/

`Scaling issue is back with casa temperature (R15).`
I forgot about this. I made a new device now - so you'll have to "configure" and select the correct one. I think this gave me alot of mess with cache or something, the values became "unavailiable", even if I restarted the integration. Suddenly they came back and worked. Hopefully just a one-off issue.

`This is not an error, but with trox "Q Min (120)" is used to control "fixed flow" with an actual value (l/s, m3/h, etc.) written to 120. This can also be achived with parameter 105 (that you now display, but this is then a "percent of nominal flow"), which I think is a less usefull (and probably less used) way to control flow.`
Yes, I tried to avoid those values with varying units, they're a headache. No change here for now.

`Would it be possible to have a "force update" button for a device? That could be really nice if you for some reason would like to make sure you have current values even if "slow update" has not run yet. I could for example change some setting (speed etc.) on my casa unit and then would all my trox make changes, due to this external event, and it would then be nice to do a "force update" if I really want to see what is going on.... An even better idea would actually be not to do a "force update", but instead switching to "fast mode" for 1-2 minutes?`
I'm not sure if using a button for this is optimal. In some other integrations I have made a specific service call that one can do (for instance from "developer tools".

`For trox; the displayed m3/h value (140) is correct since I have about 39l/s now when I check manually:`
Yeah, I do some magic in the bottom of the file. There's a function running before the first read, which manipulates the scaling.
